packages/components/ovh-shell/src/plugin/ux/components/preloader.ts

```diff
@@ -0,0 +1,58 @@
+interface IPreloaderComponent {
+  show(): void;
+  hide(): void;
+  onShow(callback: CallableFunction): void;
+  removeOnSHow(callback: CallableFunction): void;
+  onHide(callback: CallableFunction): void;
+  removeOnHide(callback: CallableFunction): void;
+}
+
+export class UxPreloader implements IPreloaderComponent {
```

Comment:
Maybe you can extends https://github.com/ovh/manager/blob/9ef42287c940b14085ae4fea9eb2377fdcfaf2eb/packages/components/ovh-shell/src/plugin/ux/components/component.ts#L13

Which handle `show` and `hide` method?